### DISCUSSION TITLE
Deleting an event in CMS not working

### DISCUSSION THREAD

**Discussion by green17** [solspace/craft-calendar#320] [Questions & Answers]
**Describe the bug or issue you're experiencing**
When trying to delete an event in the backend either in the calendar view or in the actual entry nothing happens.

Within the console, clicking delete gives a 400 error when trying to POST to below. https://domainname.com/admin/calendar/events/api/delete?site=default

**Steps to reproduce**

1. Click into calendar plugin
2. Hover over event 
3. Click delete when popup shows.

**Expected behavior**
Event should be deleted after you click confirm

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Pro 4.9.5
- Calendar Version: 5.0.4
- Calendar Edition: PRO
- Fresh Install or Upgrade: No

**Additional context**
When in 'Events' section where it shows all events in a list, if I select one and then delete via the cog at the bottom it works.

**Comment by seandelaney** [comment]
Hello,

Can you please provide some more information:

PHP version
Server type (Apache/Nginx)

Can you share what error is returned in the request?

Also, can you share what is added to the error logs when you try to delete an event?

Thanks

**Comment by green17** [comment]
PHP Version: 8.2
Server Type: Nginx

No error returned just blank screen.

Error logs don't show the events happening.

**Comment by kjmartens** [comment]
Thanks for the additional information, @green17.

In my testing, I can't duplicate this issue either. Can you provide with a little more information? 🙂 

1. Can you temporarily turn on `devMode` for your site? Doing so will generate an error on that page with a stack trace. This would help us figure out what might be causing the issue.
2. If that isn't possible, can you check Craft's logs for any errors (including stack trace) associated with this?

**Comment by green17** [comment]
@kjmartens ~This is the error:
![Screenshot 2024-08-02 at 10 38 15](https://github.com/user-attachments/assets/345ed29b-38ae-4e8e-9ec8-71314db5bc43)

**Comment by seandelaney** [comment]
@green17 

What is your server configuration? Do you have any request methods disabled/blocked?

Can you also take a screenshot of the request from the network tab in the browser console?

I want to see the request type and payload info like I have shown here:

![image](https://github.com/user-attachments/assets/207468dd-8ef1-4091-902b-1c8e96fd99d9)

Thanks

**Comment by green17** [comment]
@seandelaney 

We are using PHP Version: 8.2
Server Type: Nginx

We have no request methods disabled or blocked.

Here is the screenshot from the network tab:

![Screenshot 2024-08-12 at 15 34 13](https://github.com/user-attachments/assets/699754a4-ef71-4fd0-8dc3-085beffada01)

**Comment by seandelaney** [comment]
@green17 Can I see the response tab please?

**Comment by seandelaney** [comment]
@green17 

I'm wondering if this is a CSRF token issue (if response tab says "Bad Request Unable to verify your data submission") or if Cloudflare is blocking the request. If Cloudflare is blocking, it should show up in the Firewall Activity Events Log, along with which security setting triggered it. Have you checked there?
